src/commands/kv/bucket/sync.rs

```diff
@@ -0,0 +1,62 @@
+use std::collections::HashSet;
+use std::fs::metadata;
+use std::iter::FromIterator;
+use std::path::Path;
+
+use crate::commands::kv;
+use crate::commands::kv::bucket::directory_keys_only;
+use crate::commands::kv::bucket::upload::upload;
+use crate::commands::kv::bulk::delete::delete_bulk;
+use crate::commands::kv::key::KeyList;
+use crate::settings::global_user::GlobalUser;
+use crate::settings::target::Target;
+use crate::terminal::message;
+
+pub fn sync(
+    target: &Target,
+    user: GlobalUser,
+    namespace_id: &str,
+    path: &Path,
+) -> Result<(), failure::Error> {
+    // First get local keys
+    let local_keys_vec: Result<Vec<String>, failure::Error> = match &metadata(path) {
```

Comment:
You can actually just `?` this entire expression and get the Vec directly. (suggestion spread across three comments bc this interface does not support multiline suggestions yet)
```suggestion
    let local_keys_vec: Vec<String> = match &metadata(path) {
```

Reply:
Heck yeah!